substrate/frame/support/procedural/src/lib.rs

```diff
@@ -1157,6 +1157,36 @@ pub fn call_index(_: TokenStream, _: TokenStream) -> TokenStream {
 	pallet_macro_stub()
 }
 
+/// Each dispatchable may also be annotated with the `#[pallet::feeless_if($closure)]` attribute,
+/// which explicitly defines the condition for the dispatchable function to be feeless.
+///
+/// The arguments for the closure must be the references to arguments of the dispatchable function.
+///
+/// The closure must return `bool`.
+///
+/// ### Example
+/// ```ignore
+/// #[pallet::feeless_if(|_origin: &OriginFor<T>, something: &u32| -> bool {
+/// 		*something == 0
+/// 	})]
+/// pub fn do_something(origin: OriginFor<T>, something: u32) -> DispatchResult {
+///     ....
+/// }
+/// ```
+///
+/// Please note that this only works for signed dispatchables and requires a signed extension
+/// such as `SkipCheckIfFeeless` as defined in `pallet-skip-feeless-payment` to wrap the existing
+/// payment extension. Else, this is completely ignored and the dispatchable is still charged.
```

Comment:
Can we error at compile time instead of ignore?

Reply:
Not at this time. We will need to rethink about the overall payment flow as discussed by @kianenigma elsewhere. I think Kian already created an issue - will try to find and refer here.

Reply:
I have not created an issue, and I still think this is a minor, but pretty obvious error here. This is my original commment: https://github.com/paritytech/polkadot-sdk/issues/1725#issuecomment-1741417841 If you see agree high level, please formalize it in an issue.